src/media-gallery/media-gallery.js

```diff
@@ -158,6 +192,13 @@ component.controller = function MediaGalleryController() {
             vm.onViewAll();
         }
     };
+
+    vm.setView = function setView(view) {
+        vm.view = view;
+        if (typeof vm.onSetView === 'function') {
```

Comment:
I'm not sure you need a type check here; the '&' binding should do that for you? Please double check that. Also, do you want to pass any parameters back with onSetView, like which view type the gallery has been set to?

Reply:
I was mostly just following the convention of double checking by the other callback functions - in theory I agree with you though.

At the moment the use case I can think of for onSetView is to cache the current view in session storage. I cannot think of a particular use case for needing which view is set at the moment.